src/main/java/delete_post/use_case/DeletePostInteractor.java

```diff
@@ -0,0 +1,37 @@
+package delete_post.use_case;
+
+import delete_post.interface_adapters.DeletePostPresenter;
+import entities.User;
+
+public class DeletePostInteractor implements DeletePostInputBoundary{
+
+    private final DeletePostPresenter presenter;
+    private final DeletePostDsGateway dataAccess;
+
+    public DeletePostInteractor(DeletePostPresenter presenter, DeletePostDsGateway dataAccess){
+        this.presenter = presenter;
+        this.dataAccess = dataAccess;
+    }
+
+    @Override
+    public void delete(DeletePostRequestModel requestModel){
+        if (requestModel.getIsTimer() || requestModel.getIsAdmin() ||
+                requestModel.getUser() == (requestModel.getPostUser())){
```

Comment:
In this line, the user that created the post should be retreived from the database first